### PR TITLE
Update bicep from build templates

### PR DESCRIPTION
**Pull request by jongio**
Here are some optimizations and cleanup from project we created for build.

1. Sorted params added descriptions
2. Added support for ACA Addons with the ServiceType and ServiceBinds properties
3. Allow user to choose to not include dashboard in app insights
4. Allow user to create secrets in bulk

### CODE REVIEW

File: templates/common/infra/bicep/core/security/keyvault-secrets.bicep [hunk 1]
```diff
@@ -0,0 +1,23 @@
+param tags object = {}
+param keyVaultName string
+param secrets array = []
+
+resource keyVault 'Microsoft.KeyVault/vaults@2022-07-01' existing = {
+  name: keyVaultName
+}
+
+@batchSize(1)
```
Comment (wbreza) [inline templates/common/infra/bicep/core/security/keyvault-secrets.bicep:9]: Why are we setting batchsize to 1 here? Wouldn't we rather create them in parallel to speed up deployment?

File: templates/common/infra/bicep/core/host/container-app.bicep [hunk 2]
```diff
@@ -2,49 +2,73 @@ param name string
...
+@description('The Dapr app ID')
+param daprAppId string = containerName
+
+@description('Enable Dapr')
+param daprEnabled bool = false
+
+@description('The environment variables for the container')
 param env array = []
+
+@description('Specifies if the resource is external')
```
Comment (wbreza) [inline templates/common/infra/bicep/core/host/container-app.bicep:43]: This is part of the ingress section and actually means whether the ingress is exposed externally or not.

File: templates/common/infra/bicep/core/host/container-app-upsert.bicep [hunk 3]
```diff
@@ -2,49 +2,71 @@ param name string
...
 
-@description('CPU cores allocated to a single container instance, e.g. 0.5')
-param containerCpuCoreCount string = '0.5'
+@description('The environment variables for the container')
+param env array = []
 
-@description('Memory allocated to a single container instance, e.g. 1Gi')
-param containerMemory string = '1.0Gi'
+@description('Specifies if the resource is external')
+param external bool = true
```
Comment (wbreza) [inline templates/common/infra/bicep/core/host/container-app-upsert.bicep:61]: Same as other comment regarding whether ingress is exposed externally or not.

**Review by wbreza — APPROVED** [review]
Looks good - added a couple minor comments.

File: templates/common/infra/bicep/core/security/keyvault-secrets.bicep (hunk 1 above)
Comment (jongio) [reply templates/common/infra/bicep/core/security/keyvault-secrets.bicep:9]: Removed the file

**Comment by jongio** [comment]
Waiting on a test pass before we merge.

**Pull request merged**